### PR TITLE
Feature: Implement pagination in  conversation list screen

### LINKED ISSUE #72: Feature: Implement the pagination in conversation list

### PR DESCRIPTION
**Pull request by muhzi4u**
Fixes #72

### CODE REVIEW

File: src/actions/conversation.js [hunk 1]
```diff
@@ -31,14 +34,27 @@ export const getConversations = ({
...
   conversationStatus,
   inboxSelected,
+  pageNumber,
 }) => async dispatch => {
   dispatch({ type: GET_CONVERSATION });
   try {
+    let assignee;
+    switch (assigneeType) {
+      case 0:
+        assignee = 'me';
```
Comment (nithindavid) [inline src/actions/conversation.js]: `me`, `unassigned ` and `all` are used in more than one place, Making it constant will be better.  The same is implemented here `unassigned`.

File: src/actions/conversation.js [hunk 2]
```diff
@@ -62,6 +78,51 @@ export const getConversations = ({
...
+  conversationStatus,
+  inboxSelected,
+  pageNumber,
+}) => async dispatch => {
+  dispatch({ type: GET_MORE_CONVERSATIONS });
+  try {
+    let assignee;
+    switch (assigneeType) {
+      case 0:
+        assignee = 'me';
```
Comment (nithindavid) [inline src/actions/conversation.js]: same here, use constants

File: src/actions/conversation.js [hunk 3]
```diff
@@ -62,6 +78,51 @@ export const getConversations = ({
...
+        break;
+      case 1:
+        assignee = 'unassigned';
+        break;
+      default:
+        assignee = 'all';
+    }
+
+    const status = conversationStatus === 'Open' ? 'open' : 'resolved';
+    const inbox_id = inboxSelected && inboxSelected ? inboxSelected.id : null;
```
Comment (nithindavid) [inline src/actions/conversation.js]: camelCase ?

File: src/actions/conversation.js [hunk 4]
```diff
@@ -21,24 +21,41 @@ import {
...
+        assignee = ASSIGNEE_TYPE.UN_ASSIGNED;
+        break;
+      default:
+        assignee = ASSIGNEE_TYPE.ALL;
+    }
+
     const status = conversationStatus === 'Open' ? 'open' : 'resolved';
-    const inbox_id = inboxSelected && inboxSelected ? inboxSelected.id : null;
+    const inboxId = inboxSelected && inboxSelected ? inboxSelected.id : null;
     const apiUrl = `${API}conversations?${
```
Comment (pranavrajs) [inline src/actions/conversation.js]: You don't need to construct this url like this, you can use

```js
const url = `${API}conversations`
const params = { inbox_id: inboxId, status, assignee_type: assignee, page: pageNumber } 
return axios.get(url, params)
``

File: src/actions/conversation.js [hunk 5]
```diff
@@ -62,6 +79,51 @@ export const getConversations = ({
...
+      case 1:
+        assignee = ASSIGNEE_TYPE.UN_ASSIGNED;
+        break;
+      default:
+        assignee = ASSIGNEE_TYPE.ALL;
+    }
+
+    const status = conversationStatus === 'Open' ? 'open' : 'resolved';
+    const inboxId = inboxSelected && inboxSelected ? inboxSelected.id : null;
+    const apiUrl = `${API}conversations?${
```
Comment (pranavrajs) [inline src/actions/conversation.js]: Same why do we need an additional action to load more conversation, both cases should be handled in one action

File: src/constants/actions.js [hunk 6]
```diff
@@ -27,6 +27,10 @@ export const GET_MORE_MESSAGES = 'GET_MORE_MESSAGES';
 export const GET_MORE_MESSAGES_SUCCESS = 'GET_MORE_MESSAGES_SUCCESS';
 export const GET_MORE_MESSAGES_ERROR = 'GET_MORE_MESSAGES_ERROR';
 
+export const GET_MORE_CONVERSATIONS = 'GET_MORE_CONVERSATIONS';
```
Comment (pranavrajs) [inline src/constants/actions.js]: This actions are not required, it should be handled in getConversations itself.

**Comment by muhzi4u** [comment]
@pranavrajs Please check now

File: src/actions/conversation.js [hunk 7]
```diff
@@ -26,21 +27,46 @@ import {
...
+      default:
+        assignee = ASSIGNEE_TYPE.ALL;
+    }
+
     const status = conversationStatus === 'Open' ? 'open' : 'resolved';
-    const inbox_id = inboxSelected && inboxSelected ? inboxSelected.id : null;
-    const apiUrl = `${API}conversations?${
-      inbox_id ? `inbox_id=${inbox_id}&` : ''
-    }status=${status}&assignee_type_id=${assigneeType}`;
+    const inboxId = inboxSelected && inboxSelected ? inboxSelected.id : null;
```
Comment (pranavrajs) [inline src/actions/conversation.js]: `inboxSelected && inboxSelected ` ??

File: src/actions/conversation.js [hunk 8]
```diff
@@ -26,21 +27,46 @@ import {
...
 
-    const response = await axios.get(apiUrl);
+    const apiUrl = `${API}conversations`;
+    const params = {
+      ...(inboxId && { inbox_id: inboxId }),
```
Comment (pranavrajs) [inline src/actions/conversation.js]: There is no need for a condition here. inbox_id: inboxId will work.

**Pull request merged**